src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java

```diff
@@ -0,0 +1,58 @@
+package com.exadel.discountwebapp.vendor.service;
+
+import com.exadel.discountwebapp.vendor.entity.Vendor;
+import com.exadel.discountwebapp.vendor.mapper.VendorMapper;
+import com.exadel.discountwebapp.vendor.repository.VendorRepository;
+import com.exadel.discountwebapp.vendor.vo.VendorRequestVO;
+import com.exadel.discountwebapp.vendor.vo.VendorResponseVO;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class VendorService {
+
+    VendorMapper vendorMapper;
+    VendorRepository vendorRepository;
+
+    @Transactional(readOnly = true)
```

Comment:
propagation = Propagation.REQUIRED

Reply:
we have used for find services (readOnly = true)

Reply:
I mean: @Transactional(propagation = Propagation.REQUIRED, readOnly = true)

Reply:
changed